cluster/container.go

```diff
@@ -44,17 +44,21 @@ func StateString(state *types.ContainerState) string {
 }
 
 // FullStateString returns human-readable description of the state
-func FullStateString(state *types.ContainerState) string {
+func FullStateString(container *Container) string {
+	duration := container.Engine.DeltaDuration
+	state := container.Info.State
+
 	startedAt, _ := time.Parse(time.RFC3339Nano, state.StartedAt)
 	finishedAt, _ := time.Parse(time.RFC3339Nano, state.FinishedAt)
+
 	if state.Running {
 		if state.Paused {
-			return fmt.Sprintf("Up %s (Paused)", units.HumanDuration(time.Now().UTC().Sub(startedAt)))
+			return fmt.Sprintf("Up %s (Paused)", units.HumanDuration(time.Now().UTC().Sub(startedAt.Add(duration))))
 		}
 		if state.Restarting {
-			return fmt.Sprintf("Restarting (%d) %s ago", state.ExitCode, units.HumanDuration(time.Now().UTC().Sub(finishedAt)))
+			return fmt.Sprintf("Restarting (%d) %s ago", state.ExitCode, units.HumanDuration(time.Now().UTC().Sub(finishedAt.Add(duration))))
 		}
-		return fmt.Sprintf("Up %s", units.HumanDuration(time.Now().UTC().Sub(startedAt)))
+		return fmt.Sprintf("Up %s", units.HumanDuration(time.Now().UTC().Sub(startedAt.Add(duration))))
```

Comment:
A check is needed here to make sure "time.Now().UTC().Sub(startedAt.Add(duration)))" is not in the future. A container cannot be `up -60 seconds`, i.e., in the future.